### PR TITLE
Ehl bringup

### PR DESCRIPTION
**Pull request by shuox**
Pull request for release_2.0 branch to run EHL new board

### CODE REVIEW

**Comment by HuiHuangShi** [comment]
22 new potential violations to the coding guideline detected. Please help check. Thanks!<br><1> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func: offset:16<br> reason:User type declared but not used in code analysed. : find_iter_args<br><2> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func: offset:21<br> reason:User type declared but not used in code analysed. : dmar_iter_t<br><3> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:174<br> reason:Casting operation to a pointer. : (void* to struct acpi_table_dmar*): ( struct acpi_table_dmar * ) get_dmar_table ( )<br><4> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:174<br> reason:Casting operation on a pointer. : (void* to struct acpi_table_dmar*): ( struct acpi_table_dmar * ) get_dmar_table ( )<br><5> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:181<br> reason:No space after semi colon in for expression.<br><6> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:181<br> reason:Logical comparison of pointers.              : ptr and ptr_end<br><7> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:182<br> reason:Cast to an unrelated type. : (char* to struct acpi_dmar_header*): ( struct acpi_dmar_header * ) ptr<br><8> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:182<br> reason:Casting operation to a pointer. : (char* to struct acpi_dmar_header*): ( struct acpi_dmar_header * ) ptr<br><9> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:182<br> reason:Casting operation on a pointer. : (char* to struct acpi_dmar_header*): ( struct acpi_dmar_header * ) ptr<br><10> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:183<br> reason:Literal value requires a U suffix.           : 0<br><11> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:183<br> reason:Signed/unsigned conversion without cast. :  (unsigned short and int): dmar_header -> length<br><12> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:189<br> reason:Cast to an unrelated type. : (struct acpi_dmar_header* to struct acpi_dmar_hardware_unit*): ( struct acpi_dmar_hardware_unit * ) dmar_header<br><13> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:189<br> reason:Casting operation to a pointer. : (struct acpi_dmar_header* to struct acpi_dmar_hardware_unit*): ( struct acpi_dmar_hardware_unit * ) dmar_header<br><14> config:common filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:189<br> reason:Casting operation on a pointer. : (struct acpi_dmar_header* to struct acpi_dmar_hardware_unit*): ( struct acpi_dmar_hardware_unit * ) dmar_header<br><15> config:part_config filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:191<br> reason:Pointer not checked for null before use. : acpi_drhd<br><16> config:sharing_config filename:/hypervisor/acpi_parser/dmar_parse.c func:parse_dmar_table offset:191<br> reason:Pointer not checked for null before use. : acpi_drhd<br><17> config:part_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2236<br> reason:Pointer param should be declared pointer to const. : vlapic<br><18> config:sharing_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2236<br> reason:Pointer param should be declared pointer to const. : vlapic<br><19> config:sharing_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2246<br> reason:Pointer not checked for null before use. : pid<br><20> config:part_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2246<br> reason:Pointer not checked for null before use. : pid<br><21> config:part_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2249<br> reason:Pointer not checked for null before use. : lapic<br><22> config:sharing_config filename:/hypervisor/arch/x86/guest/vlapic.c func:vlapic_apicv_inject_pir offset:2249<br> reason:Pointer not checked for null before use. : lapic<br><br>No New Name Conflict<br><br>No New Function Declaration/Definition Mismatch<br>

File: hypervisor/hw/pci.c [hunk 1]
```diff
@@ -507,11 +507,11 @@ static void pci_parse_iommu_devscopes(struct pci_bdf_mapping_group *const bdfs_f
...
 			}
 		}
-	}
 
-	if ((plat_dmar_info.drhd_units[plat_dmar_info.drhd_count - 1U].flags & DRHD_FLAG_INCLUDE_PCI_ALL_MASK)
-			== DRHD_FLAG_INCLUDE_PCI_ALL_MASK) {
-		*drhd_idx_pci_all = plat_dmar_info.drhd_count - 1U;
+		if ((plat_dmar_info.drhd_units[drhd_index].flags & DRHD_FLAG_INCLUDE_PCI_ALL_MASK)
+				== DRHD_FLAG_INCLUDE_PCI_ALL_MASK) {
+			*drhd_idx_pci_all = drhd_index;
```
Comment (dongyaozu) [inline hypervisor/hw/pci.c:513]: This is only a style change, right?

File: hypervisor/arch/x86/guest/vlapic.c [hunk 2]
```diff
@@ -2229,21 +2229,75 @@ static void apicv_basic_inject_intr(struct acrn_vlapic *vlapic,
 }
 
 /*
- * @brief Send a Posted Interrupt to itself.
```
Comment (dongyaozu) [inline hypervisor/arch/x86/guest/vlapic.c:2232]: What is the reason of this patch? revert is not a reason.

**Comment by wenlingz** [comment]
@fyin1 please help approve this PR, thanks

**Pull request merged**